### PR TITLE
[darwin] Move common targets to common/BUILD.gn

### PR DESCRIPTION
**Pull request by cbracken**
Previously, some common Darwin framework targets existed in: 
//flutter/shell/platform/darwin/BUILD.gn

This moves all targets into:
//flutter/shell/platform/darwin/common/BUILD.gn

We also de-duplicate targets with existing targets.
* The `flutter_channels` target duplicates the existing `framework_shared` target.
* The `common` target already includes the buffer conversions translation units,
  which are only used by the iOS embedder.

No test changes since there are no semantic changes, just a restructuring of build targets.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by cbracken** [comment]
Taking a quick look to see how much pain would be involved in renaming the unittest executable to framework_shared_unittests.

**Comment by cbracken** [comment]
Pushed a second commit that renames the test target so we can add more to it later.

File: shell/platform/darwin/common/BUILD.gn [hunk 1]
```diff
@@ -56,3 +59,26 @@ source_set("framework_shared") {
 
   deps = [ "//flutter/fml" ]
 }
+
+test_fixtures("flutter_channels_fixtures") {
+  fixtures = []
+}
+
+# Unit tests for channels.
+executable("framework_shared_unittests") {
```
Comment (gaaclarke) [inline shell/platform/darwin/common/BUILD.gn:68]: Shouldn't this use the same nomenclature, "common" instead of "shared"?

**Review by gaaclarke — APPROVED** [review]

File: shell/platform/darwin/common/BUILD.gn (hunk 1 above)
Comment (cbracken) [reply shell/platform/darwin/common/BUILD.gn:68]: That's a good point. I named it after the framework_shared target, but framework_common is probably the better name. Will update. Thanks!

**Review by chinmaygarde — APPROVED** [review]

**Review by cyanglaz — APPROVED** [review]
LGTM! Thanks

**Pull request merged**